src/full/Agda/TypeChecking/Conversion.hs

```diff
@@ -609,9 +609,9 @@ compareAtom cmp t m n =
               -- since b and b' should be neutral terms, but it's a
               -- precondition for the compareAtom call to make
               -- sense.
-              equalType (El Inf $ apply tSub $ a : map (setHiding NotHidden) [bA,phi,u])
-                        (El Inf $ apply tSub $ a : map (setHiding NotHidden) [bA',phi',u'])
-              compareAtom cmp (AsTermsOf $ El Inf $ apply tSub $ a : map (setHiding NotHidden) [bA,phi,u])
+              equalType (El (Inf 0) $ apply tSub $ a : map (setHiding NotHidden) [bA,phi,u])
```

Comment:
Looks hackish.

Reply:
I agree. I'm not sure why `Inf` is needed here in the first place, but I guess it is because there is no dedicated sort for non-fibrant types like the interval / face lattice of cubical. Maybe @Saizan can comment on this.

Reply:
`Inf` is the sort of `tSub` and a few other cubical primitives.

Indeed that's because it's outside the range of sorts indexed by a `Level`, which are the ones with composition, is that still the case after this PR?

Reply:
It's still the case that `Inf` is outside of the range of sorts that are indexed by `Level`, but it is no longer disallowed to split on types in it. So it would probably be good to move the interval to its own sort if this PR is merged.